src/relentless/simulate/lammps.py

```diff
@@ -261,29 +274,57 @@ def initialize_commands(self, sim):
 
 
 class InitializeFromFile(InitializationOperation):
-    """Initialize a simulation from a LAMMPS data file.
-
-    Because LAMMPS data files only contain the LAMMPS integer types for particles,
-    you are required to specify the type map as an attribute of this operation.::
-
-        init = InitializeFromFile('lammps.data')
-        init.types = {'A': 2, 'B': 1}
-
-    Parameters
-    ----------
-    filename : str
-        The file from which to read the system data.
-
-    """
-
-    def __init__(self, filename):
+    def __init__(self, filename, format, dimension):
         self.filename = filename
+        self.format = format
+        self.dimension = dimension
 
     def initialize_commands(self, sim):
         if sim["engine"]["types"] is None:
             raise ValueError("types needs to be manually specified with a file")
-        sim[self]["_datafile"] = self.filename
-        return ["read_data {filename}".format(filename=self.filename)]
+        data_filename, dimension = self._convert_to_data_file(sim)
+        sim[self]["_datafile"] = data_filename
+        sim.dimension = dimension
+        return [f"read_data {data_filename}"]
+
+    def _convert_to_data_file(self, sim):
+        file_format = initialize.InitializeFromFile._detect_format(
+            self.filename, self.format
+        )
+        if file_format == "HOOMD-GSD":
+            if mpi.world.rank_is_root:
+                data_filename = sim.directory.temporary_file(".data")
+                with gsd.hoomd.open(self.filename) as f:
+                    frame = f[0]
+                snap = lammpsio.Snapshot.from_hoomd_gsd(frame)[0]
```

Comment:
Do we need to use the other return value to build the LAMMPS type map? I forgot about this.